pkg/datastore/datastore.go

```diff
@@ -47,8 +47,10 @@ import (
 
 type DataStore interface {
 	Find(ctx context.Context, record Record) error
+	FindSoftDeleted(ctx context.Context, record Record) error
 	FindAll(ctx context.Context, records interface{}, pagination *Pagination) error
```

Comment:
replicate this too?

Reply:
Sorry didn't get it.

Reply:
Could you please explain more? Do you mean I shouldn't replicate this function?

Reply:
FindAllSoftDeleted